monitor/utils/events.js

```diff
@@ -96,11 +97,40 @@ async function main(mode) {
       }
     })).map(normalizeEvent)
 
+    const foreignBridgeSwapped = new web3Foreign.eth.Contract([TOKENS_SWAPPED_EVENT_ABI], COMMON_FOREIGN_BRIDGE_ADDRESS)
+    const tokensSwappedEvents = (await getPastEvents(foreignBridgeSwapped, {
+      event: 'TokensSwapped',
+      fromBlock: MONITOR_FOREIGN_START_BLOCK,
+      toBlock: foreignBlockNumber
+    })).map(normalizeEvent)
+
+    // Get transfer events for each previous erc20
+    await Promise.all(
+      tokensSwappedEvents.map(async swap => {
+        const previousERC20 = new web3Foreign.eth.Contract(ERC20_ABI, swap.recipient)
+
+        const previousTransferEvents = (await getPastEvents(previousERC20, {
```

Comment:
Imagine that we have dozens of `TokensSwapped` events for the same pair of tokens. E.g. for the case when the bridge contracts will initiate the swap process every time when new SCD tokens appears on the balance. It will lead that for every instance of `TokensSwapped` the monitor will filter `Transfer` again and again for the same token contract address getting the same results. Consider to implement functionality to request `Transfer` events only for unique token contract addresses.

Reply:
Updated the code to deduplicate token contract addresses in order to no process them twice https://github.com/poanetwork/tokenbridge/pull/224/commits/a52b23cc0f4aa7e58bda9cf034d9059cf2192c67